ex16_ch4.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu May 10 15:34:22 2018
+
+@author: amyhaddad
+"""
+
+#step1: state constant
+LEGAL_DRIVING_AGE = 16
+
+#step2: get input from user on their age
+user_age = int(input("Enter your age: "))
+
+#step3: set up first conditional
+if user_age >= LEGAL_DRIVING_AGE:
+    print("You're old enough to drive legally.")
+
+#step4: set up second conditional if user enters a number less than 0
+if user_age < 0: 
+    print("Please enter a valid age.")
+
+#step5: set up conditional if user is older than 0 but less than 16
```

Comment:
If you move the check for the invalid age as the first check, then you can do an `if`-`else` instead:

```
if user_age >= LEGAL_DRIVING_AGE:
  print("You're old enough to drive legally.")
else:
  print("You're not old enough to drive legally.")
```

This is more idiomatic as it's essentially a fork in the branch, which a `if`-`else` represents well.

Reply:
@paulghaddad I made this an if-else in my update. However, if the user enters an invalid number (ie, -1) then the "else" statement is still executed. Why is this?